dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/services/jobService/data/PutJobData.kt

```diff
@@ -96,18 +105,22 @@ data class PutJobData(private val items: List<Pair<String, Path>>,
     private fun dataToDs3Objects(item: Pair<String, Path>): Iterable<Ds3Object> {
         val localDelim = item.second.fileSystem.separator
         val parent = item.second.parent
-        val ds3ObjectList = Files.walk(item.second)
-                .filter { !(Files.isDirectory(it) && Files.list(it).findAny().isPresent) }
-                .map {
-                    Ds3Object(if (Files.isDirectory(it)) {
-                        targetDir + parent.relativize(it).toString().replace(localDelim, "/") + "/"
-                    } else {
-                        targetDir + parent.relativize(it).toString().replace(localDelim, "/")
+        val paths = Files.walk(item.second).use { stream: Stream<Path> ->
```

Comment:
consider pulling this out into its own method.  Also, why return a list?  Doesn't stream have a forEach method on it?  That would avoid having to aggregate it all up into a list to then turn around and re-iterate over it.

Reply:
Oh nvm, I answered my own question.  This method side effects...